src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,418 @@
+    {
+        System.Console.Write($""{first} {other}"");
+    }
+    static void Main()
+    {
+        M(1, first: 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (10,14): error CS1744: Named argument 'first' specifies a parameter for which a positional argument has already been given
+                //         M(1, first: 2);
+                Diagnostic(ErrorCode.ERR_NamedArgumentUsedInPositional, "first").WithArguments("first").WithLocation(10, 14)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(1, first: 2)", invocation.ToString());
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol);
+        }
+
+        [Fact]
+        public void TestPositionalUnaffected2()
+        {
+            var source = @"
+class C
+{
+    static void M(int a, int b, int c = 1)
+    {
+        System.Console.Write($""M {a} {b}"");
+    }
+    static void Main()
+    {
+        M(c: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (10,9): error CS7036: There is no argument given that corresponds to the required formal parameter 'a' of 'C.M(int, int, int)'
+                //         M(c: 1, 2);
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("a", "C.M(int, int, int)").WithLocation(10, 9)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<InvocationExpressionSyntax>().ElementAt(1);
+            Assert.Equal("M(c: 1, 2)", invocation.ToString());
+            AssertEx.Equal(new[] { "void C.M(System.Int32 a, System.Int32 b, [System.Int32 c = 1])" },
+                model.GetSymbolInfo(invocation).CandidateSymbols.Select(c => c.ToTestDisplayString()));
+        }
+
+        [Fact]
+        public void TestNamedParams()
+        {
+            var source = @"
+class C
+{
+    static void M(params int[] x)
+    {
+    }
+    static void Main()
+    {
+        M(x: 1, 2);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (9,11): error CS8321: Named argument 'x' is used out-of-position but is followed by an unnamed argument
+                //         M(x: 1, 2);
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "x").WithArguments("x").WithLocation(9, 11)
```

Comment:
>Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "x").WithArguments("x").WithLocation(9, 11) [](start = 16, length = 97)

This error also feels confusing to me:
- 'x' is the first parameter
- the argument named 'x' is also first, therefor it is not used out of position

I would expect an error saying that second argument doesn't have corresponding parameter, or error CS1501: No overload for method 'M' takes 2 arguments